pkg/batch/executor.go

```diff
@@ -11,11 +11,12 @@ import (
 // dispatching new ones would start blocking.
 const RequestBufferSize = 1 << 17
 
-type Executable interface {
+type Executer interface {
 	Execute() (interface{}, error)
 }
 
-type Batchable interface {
+type BatchTracker interface {
+	// This method should be called on adding a request to an existing batch
```

Comment:
Go has weird documentation rules.  Here it should probably be
```suggestion
	// Batched is called when a request is added to an existing batch.
```